src/ImageSharp.Formats.Jpeg/JpegDecoderCore.cs

```diff
@@ -294,6 +302,9 @@ public void Decode<TColor>(Image<TColor> image, Stream stream, bool configOnly)
                             return;
                         }
 
+                        // when this is a progressive image this gets called a number of times
+                        // need to know how many times this should be called in total.
+                        mcuSet = true;
```

Comment:
I think the number depends on the individual encoding. https://msdn.microsoft.com/en-us/library/windows/desktop/ee720036(v=vs.85).aspx#_jpegprogressivedecoding